src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,628 @@
+                            $runsOn = $this->annuallyFields->getValue('runsOnThe', 'n');
+                            $month = $this->annuallyFields->getValue('month', (int) $this->start->format('m'));
+                            if (is_string($month)) {
+                                $datetime = DateTime::createFromFormat('!M', $month);
+                                if (! $datetime) {
+                                    throw new InvalidArgumentException(sprintf('Invalid month provided: %s', $month));
+                                }
+
+                                $month = (int) $datetime->format('m');
+                            }
+
+                            $rule = "FREQ=YEARLY;INTERVAL=1;BYMONTH=$month;";
+                            if ($runsOn === 'n') {
+                                $rule = new RRule($rule);
+
+                                break;
+                            }
+                        }
+
+                        $element = $this->monthlyFields;
+                        if ($customFrequency === RRule::YEARLY) {
+                            $element = $this->annuallyFields;
+                        }
+
+                        $runDay = $element->getValue('day', $element::$everyDay);
+                        $ordinal = $element->getValue('ordinal', $element::$first);
+                        $position = $element->getOrdinalAsInteger($ordinal);
+
+                        if ($runDay === $element::$everyDay) {
+                            $rule .= "BYDAY=MO,TU,WE,TH,FR,SA,SU;BYSETPOS=$position";
+                        } elseif ($runDay === $element::$everyWeekday) {
+                            $rule .= "BYDAY=MO,TU,WE,TH,FR;BYSETPOS=$position";
+                        } elseif ($runDay === $element::$everyWeekend) {
+                            $rule .= "BYDAY=SA,SU;BYSETPOS=$position";
+                        } else {
+                            $rule .= sprintf('BYDAY=%d%s', $position, $runDay);
+                        }
+
+                        $rule = new RRule($rule);
+
+                        break;
+                    default:
+                        throw new LogicException(sprintf('Custom frequency %s is not supported!', $customFrequency));
+                }
+        }
+
+        if ($rule instanceof RRule) {
+            $rule = $rule->jsonSerialize();
+        }
+
+        $values = [
+            'frequency' => $rule,
+            'start'     => $start,
+            'end'       => null
+        ];
+
+        if (parent::getValue('use-end-time', 'n') === 'y') {
+            $values['end'] = parent::getValue('end');
+        }
+
+        return $values;
+    }
+
+    public function setValue($value)
+    {
+        if (isset($value['start']) && $value['start'] instanceof DateTime) {
+            $this->setStart($value['start']);
+        }
+
+        if (isset($value['end'])) {
+            $value['use-end-time'] = 'y';
```

Comment:
Not also a call to `setEnd`? You call `setStart` above, that's why I'm asking.

Reply:
Calling `setEnd` does not have too much relevance as `setStart`. The start is not only used by this element but also by the sub elements.

Reply:
Okay, but you're not even calling `setEnd` later. Not at all? What's the purpose of `setEnd` then anyway?

Reply:
To be able to set the end time explicitly from the external, eventually.